### PR TITLE
KOGITO-9581: cli run is not showing dev mode image logs

### PR DESCRIPTION
**Pull request by ederign**
This PR:

- Log again dev mode image logs on stdout
- Refactoring docker run method

### CODE REVIEW

File: packages/kn-plugin-workflow/pkg/common/containers.go [hunk 1]
```diff
@@ -209,36 +221,92 @@ func runDockerContainer(portMapping string, path string) error {
...
 
 	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
-		fmt.Printf("Unable to start container %s", resp.ID)
+		return fmt.Errorf("\nUnable to start container %s", resp.ID)
+	}
+
+	return nil
+}
+
+func runDockerContainer(portMapping string, path string) error {
```
Comment (ederign) [inline packages/kn-plugin-workflow/pkg/common/containers.go:241]: start review by here.

File: packages/kn-plugin-workflow/pkg/common/containers.go [hunk 2]
```diff
@@ -209,36 +221,92 @@ func runDockerContainer(portMapping string, path string) error {
...
 	case <-statusCh:
+		//state of the container matches the condition, in our case WaitConditionNotRunning
 	}
 
-	out, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStdout: true})
-	if err != nil {
+	return nil
+}
+
+func processDockerLog(reader io.ReadCloser) error {
```
Comment (ederign) [inline packages/kn-plugin-workflow/pkg/common/containers.go:300]: Before this it was: 

{"status":"Pulling from kiegroup/kogito-swf-devmode","id":"1.40"}
{"status":"Digest: sha256:c5ac6f064ccfee513d85bc152491623f69d39db2a1a544fee15bd7edd73c5476"}
{"status":"Status: Image is up to date for quay.io/kiegroup/kogito-swf-devmode:1.40"}

File: packages/kn-plugin-workflow/pkg/common/containers.go [hunk 3]
```diff
@@ -209,36 +221,92 @@ func runDockerContainer(portMapping string, path string) error {
...
 	statusCh, errCh := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
+
+	//Print all container logs
+	out, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true, Follow: true})
+	if err != nil {
+		return fmt.Errorf("\nError getting container logs: %s", err)
+	}
+
+	go func() {
+		_, err := stdcopy.StdCopy(os.Stdout, os.Stderr, out)
```
Comment (ederign) [inline packages/kn-plugin-workflow/pkg/common/containers.go:282]: @AjayJagan on our version we were calling:

	out, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStdout: true})
	if err != nil {
		return err
	}

	stdcopy.StdCopy(os.Stdout, os.Stderr, out)
after the container was terminating (when we receive <-statusCh).

Now, fires a gourotin and keep copying to stdoout.

**Review by AjayJagan — APPROVED** [review]
great! @ederign

**Review by paulovmr — APPROVED** [review]

**Review by domhanak — APPROVED** [review]

**Comment by domhanak** [comment]
Sorry for multi approvals, I mistook this PR for the KOGITO-9580. I should be done with this today.

**Comment by ederign** [comment]
@domhanak I'm merging this because it blocks other PRs. If you want to suggest any change just let me know and I'll address it in a new PR.

**Comment by domhanak** [comment]
@ederign no worries, I tested it and worked well

**Pull request merged**